Lib/test/test_format.py

```diff
@@ -82,6 +82,7 @@ def test_exc(formatstr, args, exception, excmsg):
         else:
             if verbose: print('no')
             print('Unexpected ', exception, ':', repr(str(exc)))
+            raise
```

Comment:
I suggest to remove this for now, and properly rewrite this function in a separate issue. This will help to backport it.

This is an ancient code which was ported to unittest. Now we can use `subTest()` if we want to see multiple failures.

Reply:
What is the issue with backporting this change? It should work fine on 3.12 too. We can rewrite this to properly use subTest() later, but this change is small and fixes a bug that led us to miss some real test failures.

Reply:
The code will be different in `main` and other branches, therefore conflicts, which should be resolved manually.

Reply:
@serhiy-storchaka, reverted